### ISSUE TITLE
RK3588 HDR colors washed out and missing plane 'ROCKCHIP_VOP2_ESMART0'

### ISSUE THREAD

**Issue by royka1** [moonlight-stream/moonlight-qt#1076]
**Describe the bug**
When I choose HDR on the RK3588 the colors are washed out, (Colors less vibrant, looks like 8 bit colors). The TV does detect the HDR switch and recognizes it as BT.2020.

**Steps to reproduce**
To make the HDR option available I had to make a change in the DTB, and moved "cursor-win-id" from ROCKCHIP_VOP2_ESMART0 to ROCKCHIP_VOP2_CLUSTER0 , made an overlay for it, installable in Armbian with "armbian-add-overlay vop.dts". 
Start Moonlight from the GDM3 login screen or from terminal with 'QT_QPA_PLATFORM=eglfs moonlight'
After this the HDR option is available but the mouse cursor isn't visible. So it seems like the ROCKCHIP_VOP2_ESMART0 plane is not being detected.
Using the keyboard to navigate through the menus (TAB), enable HDR and start the streem.
I've tried to disable and enable the HDR option in Windows without succes.

**Affected games**
Desktop

**Other Moonlight clients**
On Android this problem doesn't occur.

**Client PC details (please complete the following information)**
Orange Pi 5 pus
Armbian
Linux orangepi5 5.10.160-legacy-rk35xx #1 SMP Mon Aug 28 01:21:24 UTC 2023  aarch64 GNU/Linux

**Server PC details (please complete the following information)**
Windows 10
Sunshine
Geforce RTX 4070 TI

**Moonlight Logs (please attach)**
[moonlightlog.log](https://github.com/moonlight-stream/moonlight-qt/files/12776662/moonlightlog.log)

**Additional context**
Anything else you think may be relevant to the issue
Rename to dts and compile/enable with 'armban-add-overlay vop.dts'
[vop.txt](https://github.com/moonlight-stream/moonlight-qt/files/12776666/vop.txt)

**Comment by kurac338** [comment]
Take a look at this thread: https://github.com/moonlight-stream/moonlight-qt/issues/974
It contains a similar discussion which was resolved a couple of weeks ago.

TL;DR: For RPi the issue was fixed, but it's still has not been officially released to the debian package repository (still not publicly available). The next official release should fix the issue for you.
If you are inpatient or want to check for yourself, you can try to compile the latest Git commit and verify how it behaves on your system.

**Comment by royka1** [comment]
@kurac338 I've built it from the Master repository so this is not the solution unfortunately.

**Comment by cgutman** [comment]
Yeah, I also have an Orange Pi 5 and looked at this for hours a few weeks back (both for moonlight-embedded and moonlight-qt), but I was unable to get it to work properly. The Rockchip DRM driver seems to be activating the 8-bit to 10-bit conversion codepath even when I'm feeding it 10-bit frames.

It also lacks proper support for the standard [COLOR_RANGE](https://drmdb.emersion.fr/properties/4008636142/COLOR_RANGE) and [COLOR_ENCODING](https://drmdb.emersion.fr/properties/4008636142/COLOR_ENCODING) plane properties, instead opting for vendor-specific [EOTF](https://drmdb.emersion.fr/properties/4008636142/EOTF) and [COLOR_SPACE](https://drmdb.emersion.fr/properties/4008636142/COLOR_SPACE) properties.

The `EOTF` and `COLOR_SPACE` properties are atomic plane properties, so we cannot use them with our non-atomic renderer. However, even if we did adopt the atomic API as I did in moonlight-embedded, the result is still unusable because the RK DRM driver has some [horrific performance issues](https://github.com/moonlight-stream/moonlight-embedded/commit/b1d111b5f796dd2bfb409106878a6f2a58205b11#r126404648) that required me to disable it for all cases where it wasn't strictly required. When I use the atomic API, latency builds up in the display pipeline until frames start being dropped, even if I request non-blocking flips.

**Comment by royka1** [comment]
Thanks for the explanation, for me it's not a problem, I'll receive an Intel N100 box tomorrow. It seems like they do a better job with their drivers, although I'll still try a few things. 
FYI to let Kodi work correctly, besides the dtb overlay I did also had to remove the line 'disable-win-move;' from the DTB, as a quick fix removed it via armbian-config -> System -> DTC. Without removing that line the video went black with every pop-up/overlay.
